src/cljs/main/broadfcui/page/workspace/method_configs/synchronize.cljs

```diff
@@ -0,0 +1,153 @@
+(ns broadfcui.page.workspace.method-configs.synchronize
+  (:require
+   [dmohs.react :as react]
+   [clojure.set :as set]
+   [clojure.string :as string]
+   [broadfcui.common.components :as comps]
+   [broadfcui.common.style :as style]
+   [broadfcui.components.modals :as modals]
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.utils :as utils]
+   ))
+
+
+(defonce ^:private sync-flag (atom false))
+
+(defn flag-synchronization []
+  (reset! sync-flag true))
+
+(defn- check-synchronization []
+  (let [val @sync-flag]
+    (reset! sync-flag false)
+    val))
+
+
+(defn- get-method-display [{:keys [namespace name snapshotId]}]
+  (str namespace "/" name " snapshot " snapshotId))
+
+(react/defc- SynchronizeModal
+  {:render
+   (fn [{:keys [props state this]}]
+     [modals/OKCancelForm
+      {:header "Synchronize Access to Method"
+       :dismiss (:dismiss props)
+       :content
+       (react/create-element
+        [:div {:style {:maxWidth 670}}
+         [:div {:style {:marginBottom "1.5rem"}}
+          "In order to allow other users of this workspace to run this method, you will need
+           to grant them access to "
+          [:b {} (get-method-display (:method props))]
+          "."]
+         (if (:customize? @state)
+           (this :-render-permission-detail)
+           (style/create-link {:text "Customize Permissions"
+                               :onClick #(swap! state assoc :customize? true)}))
+         [comps/ErrorViewer {:error (:grant-error @state)}]])
+       :ok-button [comps/Button {:text (if (:customize? @state)
+                                         "Grant Permission"
+                                         "Grant Read Permission")
+                                 :onClick #(this :-grant-permission)}]}])
+   :-render-permission-detail
+   (fn [{:keys [props locals]}]
+     [:div {}
+      [:p {} "Users currently without access:"]
+      [:table {}
+       [:tbody {}
+        (map (fn [user]
+               [:tr {}
+                [:td {:style {:verticalAlign "baseline"}} user]
+                [:td {:style {:verticalAlign "baseline"}}
+                 [:div {:style {:marginLeft "1rem"}}
+                  (style/create-identity-select
+                   {:style {:width 120}
+                    :onChange #(swap! locals assoc user (.. % -target -value))}
+                   ["Reader" "Owner" "No access"])]]])
+             (:users props))]]])
+   :-grant-permission
+   (fn [{:keys [props state locals]}]
+     (swap! state dissoc :grant-error)
+     (endpoints/call-ajax-orch
+      {:endpoint (endpoints/persist-agora-entity-acl false (:method props))
+       :payload (if (:customize? @state)
+                  (mapv (fn [[user level]] {:user user :role (string/upper-case level)}) @locals)
```

Comment:
Any users who were specified to have "no access" should just be removed from the payload altogether. In other words, send:
```
[
  { "user" : "alice@example.com", "role" : "READER"},
  { "user" : "carol@example.com", "role" : "OWNER"}
]
```
instead of:
```
[
  { "user" : "alice@example.com", "role" : "READER"},
  { "user" : "bob@example.com", "role" : "NO ACCESS"},
  { "user" : "carol@example.com", "role" : "OWNER"}
]
```
first, it's a barely more efficient payload. But more importantly, sending an explicit "no access" ACL for a user triggers a weirdness in agora, where agora actually inserts the no-access ACL and that user appears in the permissions list (as "no access").